libmachine/provision/utils.go

```diff
@@ -190,7 +190,7 @@ func ConfigureAuth(p Provisioner) error {
 		return err
 	}
 
-	if err := p.Service("docker", serviceaction.Start); err != nil {
+	if err := p.Service("docker", serviceaction.Restart); err != nil {
```

Comment:
we sure doing a restart will work with older versions? any reason to gate this behind a v20.10.x check?

Reply:
The patch have been tested successfully with https://releases.rancher.com/install-docker/19.03.sh and https://releases.rancher.com/install-docker/20.10.sh at ubuntu 18.04 and 20.04. 

The docker daemon should be also restarted on older versions. For some reason, on some cloud providers (same config/os works fine at aws but not at do)  systemd is not stoping docker daemon properly before start. The patch is assuring the docker daemon is restarted (instead of started) once it's reconfigured.

Reply:
Tested succesfully using https://releases.rancher.com/install-docker:
- 18.09.sh: ubuntu 16.04 18.04 aws and do
- 19.03.sh: ubuntu 18.04 20.04 aws and do
- 20.10.sh: ubuntu 18.04 20.04 aws and do